shared/src/main/scala/coop/rchain/casper/pCasper/sim/Simulation.scala

```diff
@@ -384,4 +390,33 @@ object Simulation {
           else newNet.asRight                       // Final value
         res.pure[F]
     }
+
+  final case class FringeProcessor(fringes: Map[Sender, Vector[Set[Msg]]]) {
+    def addSenderFringe(sender: Sender, fringe: Set[Msg]): FringeProcessor = {
+      // Add new fringe
+      val newFringes = fringes + fringes
+        .get(sender)
+        .map(v => v :+ fringe)
+        .map(sender -> _)
+        .getOrElse(sender -> Vector(fringe))
+
+      // Check if new fringe is consistent with existing fringes
+
+      // intersectingFringes is a sequence of messages which each sender has
+      // Example:
+      //  Sender1: [0 - 1 - 2] - 3 - 4
+      //  Sender2: [0 - 1 - 2]
+      //  Sender3: [0 - 1 - 2] - 3
+      val minFringeLength     = newFringes.values.map(_.length).min
```

Comment:
I think it would be better to store an index for a fringe, make `Vector[Set[Msg]` -> `indexedSeq[Set[Msg]`, so you have to only get fringe option by index for all senders in `fringes`, and check if they are absent or the same that you are going to insert.

Reply:
[Done](https://github.com/nzpr/rchain/pull/6/commits/e7c5f6c6f55732f4907c010e199609f505d0f799).